src/azure-cli/azure/cli/command_modules/role/_msgrpah/_graph_client.py

```diff
@@ -364,6 +357,24 @@ def _filter_to_query(filter):
     return ''
 
 
+def _get_user_url(id_or_upn):
+    # Correctly handle $ and # in upn according to
+    # https://docs.microsoft.com/en-us/graph/api/user-get
+    # https://docs.microsoft.com/en-us/graph/known-issues#users
+
+    # UPN
+    if '@' in id_or_upn:
+        from urllib.parse import quote
+        encoded_upn = quote(id_or_upn, safe='')
+
+        if id_or_upn.startswith('$'):
+            return f"/users('{encoded_upn}')"
```

Comment:
Here's a little bit difference from the original implementation. Is it possible that `id_or_upn` doesn't contain `@` but starts with `$`?

Reply:
Nope. If `id_or_upn` doesn't contain `@`, it can only be a GUID.